### PR TITLE
feat: [PL-31102]: Add new Redis Batch iterator support for all iterators in manager

### PR DESCRIPTION
**Pull request by Raghav-Murali**
## Describe your changes

This PR extends the support for the new Redis batch mode iterator for all the iterators in the manager.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43094)

### CODE REVIEW

**Comment by Raghav-Murali** [comment]
trigger checkstylecodeformat, messagecheck

**Comment by Raghav-Murali** [comment]
trigger ti0

**Comment by Raghav-Murali** [comment]
trigger ti1

File: 400-rest/src/main/java/software/wings/scheduler/marketplace/gcp/GCPBillingHandler.java [hunk 1]
```diff
@@ -63,7 +66,34 @@ protected void createAndStartIterator(
...
+    iterator = MongoPersistenceIterator.<GCPBillingJobEntity, MorphiaFilterExpander<GCPBillingJobEntity>>builder()
+                   .mode(ProcessMode.REDIS_BATCH)
+                   .clazz(GCPBillingJobEntity.class)
+                   .fieldName(GCPBillingJobEntityKeys.nextIteration)
+                   .targetInterval(targetInterval)
+                   .acceptableNoAlertDelay(ACCEPTABLE_NO_ALERT_DELAY)
+                   .executorService(executor)
+                   .semaphore(semaphore)
+                   .handler(this)
+                   .entityProcessController(new AccountStatusBasedEntityProcessController<>(accountService))
```
Comment (gn-harness) [inline 400-rest/src/main/java/software/wings/scheduler/marketplace/gcp/GCPBillingHandler.java:89]: where is this coming from?

**Review by Raghav-Murali — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43094)* status: 32 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/scheduler/marketplace/gcp/GCPBillingHandler.java` line 89 at r1](https://reviewable.io/reviews/harness/harness-core/43094#-NMfhoUE64Shh5bmKp7z-r1-89:-NNO45KUBaASxSFfSr2u:b-k6bq5q) ([raw file](https://github.com/harness/harness-core/blob/2b048863be35381387e84bffbcabeff92f1f6813/400-rest/src/main/java/software/wings/scheduler/marketplace/gcp/GCPBillingHandler.java#L89)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

where is this coming from?
</blockquote></details>

The accountService comes from Injector. The rest like executor and semaphore are instantiated in the createAndStartRedisBatchIterator using the config from the iterator_config.json

**Review by gn-harness — APPROVED** [review]
> Reviewed 32 of 32 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43094)* status: 32 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by Raghav-Murali** [comment]
trigger feature-build

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 32 of 32 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43094)* status: 1. unresolved discussions.

**Review by gn-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43094)* status: :shipit: complete! ✓

**Pull request merged**